TCAT/AppDelegate.swift

```diff
@@ -24,6 +24,27 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
         
+        // Update shortcut items
+        var shortcutItems = [UIApplicationShortcutItem]()
+        let favorites = SearchTableViewManager.shared.retrieveRecentPlaces(for: Constants.UserDefaults.favorites)
+        for itemType in favorites {
+            switch itemType {
+            case .busStop(let busStop):
+                let data = NSKeyedArchiver.archivedData(withRootObject: busStop)
+                let placeInfo: [AnyHashable: Any] = ["place": data]
+                let shortcutItem = UIApplicationShortcutItem(type: busStop.name, localizedTitle: busStop.name, localizedSubtitle: nil, icon: UIApplicationShortcutIcon(type: .location), userInfo: placeInfo)
+                shortcutItems.append(shortcutItem)
+            case .placeResult(let placeResult):
+                let data = NSKeyedArchiver.archivedData(withRootObject: placeResult)
+                let placeInfo: [AnyHashable: Any] = ["place": data]
+                let shortcutItem = UIApplicationShortcutItem(type: placeResult.name, localizedTitle: placeResult.name, localizedSubtitle: nil, icon: UIApplicationShortcutIcon(type: .location), userInfo: placeInfo)
+                shortcutItems.append(shortcutItem)
+            case .cornellDestination: break
+            case .seeAllStops: break
+            }
+        }
```

Comment:
It's best to leave this function as small as possible and outsource things like this into its own function, because it can get crowded and it helps with maintenance and readability. Make a function called `updateShortcutItems`, declare it outside the function with a `// MARK: App Shortcut` above, and call it within the delegate function.